aptos-move/framework/aptos-token-objects/sources/collection.move

```diff
@@ -324,6 +427,60 @@ module aptos_token_objects::collection {
         MutatorRef { self: object::object_address(&object) }
     }
 
+    public fun upgrade_to_concurrent(
+        ref: &ExtendRef,
+    ) acquires FixedSupply, UnlimitedSupply {
+        let metadata_object_address = object::address_from_extend_ref(ref);
+        let metadata_object_signer = object::generate_signer_for_extending(ref);
+        assert!(features::concurrent_assets_enabled(), error::invalid_argument(ECONCURRENT_NOT_ENABLED));
+
+        if (exists<FixedSupply>(metadata_object_address)) {
+            let FixedSupply {
+                current_supply,
+                max_supply,
+                total_minted,
+                burn_events,
+                mint_events,
+            } = move_from<FixedSupply>(metadata_object_address);
+
+            event::destroy_handle(burn_events);
+            event::destroy_handle(mint_events);
+
+            let supply = ConcurrentSupply {
```

Comment:
Next 7-8 lines look the same for both branches. How about moving them to after the if/else?

Reply:
moving what is the same, creation of supply object is different though